File: shared/types/knowledge.ts
export interface Document {
  id: string;
  filename: string;
  file_type: "pdf" | "docx" | "md" | "txt";
  size_bytes: number;
  chunk_count: number;
  uploaded_at: string;
}

export interface KnowledgeChunk {
  id: string;
  document_id: string;
  content: string;
  embedding?: number[];
  metadata: Record<string, unknown>;
}

export interface SearchQuery {
  query: string;
  top_k?: number;
  filters?: Record<string, string>;
}

export interface SearchResult {
  chunk: KnowledgeChunk;
  score: number;
  document: Document;
}

File: backend/app/models/knowledge.py
from pydantic import BaseModel
from typing import Optional, List, Dict


class Document(BaseModel):
    id: str
    filename: str
    file_type: str
    size_bytes: int
    chunk_count: int = 0
    uploaded_at: str


class KnowledgeChunk(BaseModel):
    id: str
    document_id: str
    content: str
    embedding: Optional[List[float]] = None
    metadata: Dict = {}


class SearchQuery(BaseModel):
    query: str
    top_k: int = 5
    filters: Optional[Dict[str, str]] = None


class SearchResult(BaseModel):
    chunk_id: str
    content: str
    score: float
    document_id: str
